src/libp2p/Libp2p.Protocols.TLS/TLS.cs

```diff
@@ -0,0 +1,408 @@
+        {
+            _logger?.LogTrace("Sslstream initialized.1");
+            await sslStream.AuthenticateAsServerAsync(serverAuthenticationOptions);
+            _logger?.LogTrace("Sslstream initialized.2");
+        }
+        catch (Exception ex)
+        {
+            _logger?.LogError("An error occurred during TLS authentication: {Message}", ex.Message);
+            _logger?.LogDebug("Exception details: {StackTrace}", ex.StackTrace);
+            throw;
+        }
+
+        if (sslStream.NegotiatedApplicationProtocol == SslApplicationProtocol.Http2)
+        {
+            _logger?.LogDebug("HTTP/2 protocol negotiated");
+        }
+        else if (sslStream.NegotiatedApplicationProtocol == SslApplicationProtocol.Http11)
+        {
+            _logger?.LogDebug("HTTP/1.1 protocol negotiated");
+        }
+        else if (sslStream.NegotiatedApplicationProtocol == SslApplicationProtocol.Http3)
+        {
+            _logger?.LogDebug("HTTP/3 protocol negotiated");
+        }
+        else
+        {
+            _logger?.LogDebug($"Protocol negotiated!");
+        }
+
+        IChannel upChannel = channelFactory.SubListen(context);
+
+        LastNegotiatedApplicationProtocol = sslStream.NegotiatedApplicationProtocol;
+
+
+        await ExchangeData(sslStream, upChannel, _logger);
+
+    }
+
+
+    private static bool VerifyRemoteCertificate(Multiaddress remotePeerAddress, X509Certificate certificate) =>
+             CertificateHelper.ValidateCertificate(certificate as X509Certificate2, remotePeerAddress.Get<P2P>().ToString());
+
+    public class ChannelStream(IChannel chan, ILogger<TlsProtocol> logger) : Stream
+    {
+        private bool _disposed = false;
+        private bool _canRead = true;
+        private bool _canWrite = true;
+
+        public override bool CanRead => _canRead;
+        public override bool CanSeek => false;
+        public override bool CanWrite => _canWrite;
+        public override long Length => throw new Exception();
+
+        public override long Position
+        {
+            get => 0;
+            set => throw new NotSupportedException();
+        }
+
+        public override void Flush() { }
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return Read(buffer.AsSpan(offset, count));
+        }
+        public override int Read(Span<byte> buffer)
+        {
+            logger.LogInformation("READ 1");
+            if (buffer is { Length: 0 } && _canRead)
+            {
+                return 0;
+            }
+            logger.LogInformation("READ 2");
+
+            var result = chan.ReadAsync(buffer.Length, ReadBlockingMode.WaitAny).Result;
+
+            logger.LogInformation($"READ 3 {result.Result}");
+
+            if (result.Result != IOResult.Ok)
+            {
+                _canRead = false;
+                return 0;
+            }
+            result.Data.CopyTo(buffer);
+            logger.LogInformation("READ 4");
```

Comment:
It might be better to introduce more meaningfull logs or remove these ones at all in this class